misk/src/main/kotlin/misk/web/extractors/PathPatternParameterExtractor.kt

```diff
@@ -13,9 +14,14 @@ import kotlin.reflect.full.findAnnotation
  * and returns it as a [String]. If the parameter name doesn't occur in [pathPattern], returns null.
  */
 object PathPatternParameterExtractorFactory : ParameterExtractor.Factory {
-    override fun create(parameter: KParameter, pathPattern: PathPattern): ParameterExtractor? {
-        val pathParamAnnotation = parameter.findAnnotation<PathParam>()
-        val parameterName = pathParamAnnotation?.value ?: parameter.name
+    override fun create(function: KFunction<*>, parameter: KParameter,
+            pathPattern: PathPattern): ParameterExtractor? {
+        val pathParamAnnotation = parameter.findAnnotation<PathParam>() ?: return null
+        val parameterName = when (pathParamAnnotation.value.isBlank()) {
```

Comment:
Is this an if clause disguised as a when clause?

Reply:
Yeah, stylistically it seems like we prefer `when` over `if` evaluating as an expression? I've mostly seen our code doing `when`s even with just two conditions. I'm cool changing this to if though, it's more correct.